clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java

```diff
@@ -70,6 +75,17 @@ public AbstractConfig(ConfigDef definition, Map<?, ?> originals) {
         this(definition, originals, true);
     }
 
+    /**
+     * Called directly after user configs got parsed (and thus default values got set).
+     * This allows to change default values for "secondary defaults" if required.
+     *
+     * @param parsedValues unmodifiable map of current configuration
+     * @return a map of updates that should be applied to the configuration (will be validated to prevent bad updates)
+     */
+    protected Map<String, Object> postProcessParsedConfig(Map<String, Object> parsedValues) {
```

Comment:
One more thing to add here: with EOS, for consumer/producer config you approach is the way we go -- be we also have Streams config `COMMIT_INTERVAL_MS` that gets a different default value for EOS, and for this case, we need to set the value correctly from the beginning on.